### PR TITLE
Fixed service sorting in catalog plugin on acquire

### PR DESCRIPTION
**Pull request by HendrikPeilke**
This pull request fixes a mistake inside the catalog plugin making it choose the worst service out of the list of possible instead of the best. Also added a test.

### CODE REVIEW

File: plugins/org.locationtech.udig.catalog.tests/plugin.xml [hunk 1]
```diff
@@ -12,6 +12,16 @@
             class="org.locationtech.udig.catalog.tests.DummyMultiResourceServiceExtension"
             id="MultiDummy"
             name="MultiDummy"/>
+      <service
```
Comment (HendrikPeilke) [inline plugins/org.locationtech.udig.catalog.tests/plugin.xml]: added two services for the test

File: plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/LessInterestingService.java [hunk 2]
```diff
@@ -0,0 +1,77 @@
+package org.locationtech.udig.catalog.tests.internal;
```
Comment (HendrikPeilke) [inline plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/LessInterestingService.java]: a less interesting test service, just for the test

File: plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/CatalogImplTest.java [hunk 3]
```diff
@@ -57,4 +64,11 @@ public void testUrlEquals() throws Exception {
     public void testAquire() {
         
     }
+    
+    @Test
+    public void testServiceComparison() throws Exception {
```
Comment (HendrikPeilke) [inline plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/CatalogImplTest.java]: test acquires a service just by giving the url. The two test services match that request an the better one should be chosen.

File: plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/MoreInterestingService.java [hunk 4]
```diff
@@ -0,0 +1,77 @@
+package org.locationtech.udig.catalog.tests.internal;
```
Comment (HendrikPeilke) [inline plugins/org.locationtech.udig.catalog.tests/src/org/locationtech/udig/catalog/tests/internal/MoreInterestingService.java]: a more interesting test service, just for the test

File: plugins/org.locationtech.udig.catalog/src/org/locationtech/udig/catalog/internal/CatalogImpl.java [hunk 5]
```diff
@@ -1155,10 +1155,12 @@ public int compare( IService o1, IService o2 ) {
                     double metric1 = info1.getMetric();
                     double metric2 = info2.getMetric();
                     
+                    // needs to be sorted in descending order, so that the most matching
```
Comment (HendrikPeilke) [inline plugins/org.locationtech.udig.catalog/src/org/locationtech/udig/catalog/internal/CatalogImpl.java]: sorting got mixed up (look at the javadoc for IServiceInfo.getMetric(), it says 1, if the request matched all parameters and 0 if not).

**Comment by jesseeichar** [comment]
looks ok for me.  any other comments?

**Comment by HendrikPeilke** [comment]
Can someone please explain why the Eclipse validation says:
- The following users do not have valid CLAs: hendrik.peilke@ibykus.com
- The following users have invalid Signed-off-by footers: hendrik.peilke@ibykus.com

I did submit the Eclipse CLA (button click inside my eclipse profile page). 

Thanks.

**Comment by fgdrf** [comment]
Hendrik, Eclipse validation checks if you have committed with the signed option ("-s"), e.g 
`git commit -s -m "a message"`
Just add another commit with minor changes...

However, fix looks fine. Thanks for adding a test too!

**Comment by HendrikPeilke** [comment]
I made the test more compatible with future tests by making the test services only react on the test url. Also got Eclipse validation to work after putting my github ID into my eclipse account and using the -s flag. Thanks Frank.

**Comment by fgdrf** [comment]
After merging changes into master the buildjob marked the following test as failed:

* org.locationtech.udig.project.tests.internal.commands.CreateMapCommandTest.testWithoutProjectWithoutName  

Is it because of the two additional Catalog services in plugins/org.locationtech.udig.catalog.tests/plugin.xml?

Do you have any ideas?

**Comment by HendrikPeilke** [comment]
Yes, I should have checked the id inside createService instead of createParams. I'm fixing it right now.

**Pull request merged**